examples/khr-3hv/khr-3hv_continuous/controllers/supervisorsManager/khr3hvEnv.py

```diff
@@ -0,0 +1,206 @@
+                       "Globan distance walekd": GLOBAL_DISTANCE
+                       })
+
+        motorPos.extend([i for i in self.motorPositionArr])
+        # Shift the last 10 observations by one, on the time window of 5
+        self.obs = shift(self.obs, -self.numObs, cval=np.NaN)
+        self.obs[-self.numObs:] = motorPos
+
+        return np.array(self.obs)
+
+    def get_reward(self, action):
+        """
+        Calculate the reward of the agent. The reward award agent when moving forward on the x-axis.
+        Also, we are using a second term so the agent moves further away from his previous position.
+        :return: Reward value
+        :rtype: float
+
+        """
+        reward = 2.5 * \
+            self.robot.getPosition()[0] + \
+            self.robot.getPosition()[0] - self.prev_pos
+
+        if self.counterLoging % self.save_freq == 0 and self.train and not self.useRay:
+            wandb.log({"reward": reward,
+                       "reward-1term-weight-pos": 2.5 * self.robot.getPosition()[0],
+                       "reward-2term-diff-possition": self.robot.getPosition()[0] - self.prev_pos
+                       })
+
+        self.prev_pos = self.robot.getPosition()[0]
+        return reward
+
+    def is_done(self):
+        """
+        This method checks the termination criteria for each episode. 
+        If the criteria are satisfied return True otherwise it returns False.
+        :return: Termination criteria
+        :rtype: Boolean
+
+        """
+        # Agent fall.
+        if self.robot.getPosition()[2] < 0.5:
+            return True
+        # Agent walked out of the box
+        if self.robot.getPosition()[0] > 15:
+            return True
+        return False
+
+    def get_default_observation(self):
+        """
+        Simple implementation returning the default observation which is a zero 
+        vector in the shape of the observation space.
+        :return: Starting observation zero vector
+        :rtype: numpy array
+        """
+        self.prev_pos = 0
+        self.obs = np.zeros(self.t * self.numObs)
+        self.distance_walk = 0
+
+        return np.zeros(self.observation_space.shape[0])
+
+    def apply_action(self, action):
+        """
+        This method uses the action list provided, which contains the next action to be 
+        executed by the robot.It contains a float number denoting the action. 
+        The corresponding motorPosition value is applied at each motor.
+        :param action: The list that contains the action value
+        :type action: list of float
+        """
+        motorIndexes = [0, 1, 2, 3, 4, 5]
+
+        for i, ac in zip(motorIndexes, action):
+            self.motorPositionArr[i] += ac
+            self.motorList[i].setVelocity(self.motor_velocity)
+            self.motorList[i].setPosition(ac)
+
+    def setup_agent(self):
+        """
+        This method initializes the 17 (all) or 6 (leg)  motors, 
```

Comment:
There doesn't seem to be a way to initialize all or leg motors apart from going inside the util function and hard coding it there. I don't know if there's a reason to be able to choose between the two. 
```suggestion
        This method initializes the motors, 
```

Reply:
@tsampazk Arm swing is a major part of our balance and control while walking, so I think that is the reason to choose between these two. As for hard-coding, currently he controls those motors in its leg only.
I think we can keep `name_type='all'` since we are going to fix the convergence issues.
```python
    def getMotorNames(name_type='all'):
        if name_type == 'all':
            return ['Head', 'LeftAnkle', 'LeftArm', 'LeftCrus',
                    'LeftElbow', 'LeftFemur', 'LeftFemurHead1',
                    'LeftFemurHead2', 'LeftFoot', 'LeftForearm',
                    'LeftShoulder', 'RightAnkle', 'RightArm',
                    'RightCrus', 'RightElbow', 'RightFemur',
                    'RightFemurHead1', 'RightFemurHead2',
                    'RightFoot', 'RightForearm', 'RightShoulder',
                    'Waist']
        elif name_type == 'legs':
            return ['LeftAnkle', 'LeftCrus', 'LeftFemur',
                    'RightAnkle', 'RightCrus', 'RightFemur']
```
I think I can do some modifications and make it soft-coding. What do you think?

Reply:
Yes we could at least make an argument through the env or similar. 

If we indeed end up using the arms too, we should keep it this way and remove the legs only stuff. If we don't, i think we should remove it altogether because it will just be an unused piece of code, and keep the legs. Maybe even simplify it further by removing these functions and adding the list of names wherever it is used, because i think it gets used only once. Generally, i'd like to keep the finalized example as simple as possible.

Reply:
To further clarify my reasoning, i could see an example having a top level argument that changes the action space between "legs + arms" and "legs-only", meaning that a lot of things would change based on that argument. This would of course lead to pretty different behaviours and training.

The way it is now, we have some additional code or commented code sitting around, not really giving the option to users to easily change it.

Reply:
> Generally, i'd like to keep the finalized example as simple as possible.

I agree with you!

Reply:
@tsampazk Any further suggestions for the following? 😄
https://github.com/KelvinYang0320/deepworlds/blob/6f8cb02af17e9d04b11570b51168339858858fee/examples/khr-3hv/khr-3hv_continuous/controllers/robot_supervisor_manager/KHR3HV_env.py#L174-L192

Reply:
That looks good @KelvinYang0320, thanks!